File: cursor/backend/app/schemas/todo.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class TodoBase(BaseModel):
    title: str
    description: Optional[str] = None
    completed: bool = False


class TodoCreate(BaseModel):
    title: str
    description: Optional[str] = None


class TodoUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    completed: Optional[bool] = None


class TodoResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    description: Optional[str] = None
    completed: bool
    created_at: datetime
    updated_at: datetime

File: cursor/frontend/src/types/todo.js
/**
 * @typedef {Object} Todo
 * @property {number} id
 * @property {string} title
 * @property {string|null} description
 * @property {boolean} completed
 * @property {string} created_at
 * @property {string} updated_at
 */

/**
 * @typedef {Pick<Todo, 'title'|'description'>} TodoCreate
 */

/**
 * @typedef {Partial<Pick<Todo, 'title'|'description'|'completed'>>} TodoUpdate
 */

export {}
